extension/src/pipeline/quickPick.ts

```diff
@@ -79,36 +95,79 @@ const getFieldOptions = (data: UnknownValue): string[] => {
     : []
 }
 
-export const pickPlotConfiguration = async (): Promise<
-  PlotConfigData | undefined
-> => {
-  const file = await pickDataFile()
+const showNotEnoughKeysToast = () =>
+  Toast.showError(
+    'The requested file(s) does not contain enough keys (columns) to generate a plot. Does the file or files follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+  )
+
+const getFieldsFromDataFiles = (
+  dataArr: { data: UnknownValue; file: string }[]
+) => {
+  const keys: {
+    [file: string]: string[]
+  } = {}
+  let keysAmount = 0
+
+  for (const { file, data } of dataArr) {
+    const fields = getFieldsFromValue(data)
+
+    if (fields.length === 0) {
+      void showNotEnoughKeysToast()
+      return
+    }
+
+    keysAmount += fields.length
+
+    if (!keys[file]) {
+      keys[file] = []
+    }
+    keys[file].push(...fields)
+  }
+
+  if (keysAmount < 2) {
+    void showNotEnoughKeysToast()
+    return
+  }
+
+  return keys
+}
+
+export const pickPlotConfiguration = async (
+  cwd: string
+): Promise<PlotConfigData | undefined> => {
+  const files = await pickDataFiles()
 
-  if (!file) {
+  if (!files) {
     return
   }
 
-  const data = await loadDataFile(file)
+  const fileExts = new Set(files.map(file => getFileExtension(file)))
 
-  if (!data) {
-    return Toast.showError(
-      'Failed to parse the requested file. Does the file contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
-    )
+  if (fileExts.size > 1) {
+    return Toast.showError('Files must of the same type.')
   }
 
-  const keys = getFieldOptions(data as UnknownValue)
+  const filesData = await loadDataFiles(files)
 
-  if (keys.length < 2) {
+  if (!filesData) {
     return Toast.showError(
-      'The requested file does not contain enough keys (columns) to generate a plot. Does the file follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+      'Failed to parse the requested file(s). Does the file or files contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
```

Comment:
It feels a little weird to have both `file(s)` and `file or files` so close to one another. I'd stick with only one formulation.

Reply:
Used `file or files` since the grammar looked off with `file(s)` :)